app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -34,12 +34,14 @@ EOF
 register_host() {
   curl --cacert $SSL_CA_CERT --request POST <%= foreman_server_url %>/register \
        <%= headers.join(' ') %> \
-       -d '{ "host": { "name": "'$(hostname --fqdn)'", "build": "false", "managed": "false"
-       <%= ", \"organization_id\": \"#{@organization.id}\"" if @organization -%>
-       <%= ", \"location_id\": \"#{@location.id}\"" if @location -%>
-       <%= ", \"hostgroup_id\": \"#{@hostgroup.id}\"" if @hostgroup -%>
-       <%= ", \"operatingsystem_id\": \"#{@operatingsystem.id}\"" if @operatingsystem -%>
-       }}'
+       --data "host[name]=$(hostname --fqdn)" \
+       --data "host[build]=false" \
+       --data "host[managed]=false" \
+       <%= "--data \"host[organization_id]=#{@organization.id}\"" if @organization -%>
+       <%= "--data \"host[location_id]=#{@location.id}\"" if @location -%>
+       <%= "--data \"host[hostgroup_id]=#{@hostgroup.id}\"" if @hostgroup -%>
+       <%= "--data \"host[operatingsystem_id]=#{@operatingsystem.id}\"" if @operatingsystem -%>
```

Comment:
I'd avoid the double quoting and use single quotes
```suggestion
       <%= "--data 'host[organization_id]=#{@organization.id}'" if @organization -%>
       <%= "--data 'host[location_id]=#{@location.id}'" if @location -%>
       <%= "--data 'host[hostgroup_id]=#{@hostgroup.id}'" if @hostgroup -%>
       <%= "--data 'host[operatingsystem_id]=#{@operatingsystem.id}'" if @operatingsystem -%>
```

This also avoids any potential shell interpretation of variables or other syntax while also making it easier to read. Same below for the other curl request.